File: Menza-WA/src/main/java/hr/tvz/zr/menzawa/model/Menza.java
package hr.tvz.zr.menzawa.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "MENZE")
public class Menza {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Column()
    private String naziv;
    private String adresa;
    private String info;
    @Column(name = "radno_vrijeme")
    private String radnoVrijeme;
    private String meni;
    @JsonManagedReference
    @OneToMany(mappedBy = "menza", fetch = FetchType.EAGER)
    private List<Blagajna> blagajne;
    private Integer guzva;
}

File: Androidmenza/app/src/main/java/hr/tvz/android_menza/Menza.kt
package hr.tvz.android_menza

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.util.*

@Parcelize
data class Menza(@SerializedName("id") var id: UUID,
                 @SerializedName("naziv") var naziv: String,
                 @SerializedName("adresa") var adresa:String,
                 @SerializedName("info") var info:String,
                 @SerializedName("radnoVrijeme") var radnoVrijeme: String,
                 @SerializedName("meni") var meni:String,
                 @SerializedName("blagajne") var blagajne: kotlin.collections.List<Blagajna>): Parcelable {
}